packages/react/popover/src/Popover.tsx

```diff
@@ -261,13 +261,12 @@ const PopoverContentImpl = React.forwardRef((props, forwardedRef) => {
           // (closed !== unmounted when animating out)
           trapped={trapFocus && context.open}
           onMountAutoFocus={onOpenAutoFocus}
-          onUnmountAutoFocus={(event) => {
-            if (!disableOutsidePointerEvents && isPointerDownOutsideRef.current) {
-              event.preventDefault();
-            } else {
-              onCloseAutoFocus?.(event);
-            }
-          }}
+          onUnmountAutoFocus={composeEventHandlers(props.onCloseAutoFocus, (event) => {
+            const isPointerOutside = isPointerDownOutsideRef.current;
+            const isAllowedPointerOutside = isPointerOutside && !disableOutsidePointerEvents;
+            if (!isAllowedPointerOutside) context.triggerRef.current?.focus();
+            event.preventDefault();
```

Comment:
nit:

```suggestion
            event.preventDefault();
            if (!isAllowedPointerOutside) context.triggerRef.current?.focus();
```

I think generally when we take over an event, we prevent default first and then do our action. I know it doesn't matter technically but I feel like it reads better in terms of the intent.